harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/database/DatabaseMetadataRetriever.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.util.StringUtils;
+
+import java.sql.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+class DatabaseMetadataRetriever {
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    protected Logger log;
+
+    /**
+     * Constructor.
+     *
+     * @param connectionString An example of server string in case of Oracle
+     *                         is: "jdbc:oracle:thin:@84.123.79.19:1521:orcl".
+     * @param username the username to connect to the database.
+     * @param password the password to connect to the database.
+     */
+    public DatabaseMetadataRetriever(String driverName, String connectionString, String username, String password, Logger log) {
+
+        try {
+            log.debug("Getting database connection (via JDBC)");
+
+            BasicDataSource dataSource = new BasicDataSource();
+            dataSource.setDriverClassName(driverName);
+            dataSource.setUrl(connectionString);
+            dataSource.setUsername(username);
+            dataSource.setPassword(password);
+            // Test the connection config getting a connection and closing it.
+            dataSource.getConnection().close();
+
+            jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+
+            this.log = log;
+        } catch (SQLException x) {
+            log.error("Error getting database connection", x);
+
+            throw new ExceptionInInitializerError(new DatabaseMetadataRetrieverException("Exception in getting database connection: can not connect to the database", x));
+        }
+    }
+
+    protected NamedParameterJdbcTemplate getJdbcTemplate() {
+        return this.jdbcTemplate;
+    }
+
+
+    /**
+     * Retrieves and process each metadata with the harvester aligner.
+     *
+     * @param cancelMonitor
+     * @param params
+     * @param aligner
+     * @throws Exception
+     */
+    public void processMetadata(AtomicBoolean cancelMonitor, DatabaseHarvesterParams params, DatabaseHarvesterAligner aligner) throws Exception {
+        String metadataTable = params.getTableName();
+        String columnName = params.getMetadataField();
+        String filterField = params.getFilterField();
+        String filterValue = params.getFilterValue();
+
+        String sqlQuery;
+        SqlParameterSource param = new MapSqlParameterSource();
+
+        if (StringUtils.hasLength(filterField) && StringUtils.hasLength(filterValue)) {
+            sqlQuery = String.format("SELECT %s FROM %s WHERE %s LIKE :filter", columnName, metadataTable, filterField);
+            param = new MapSqlParameterSource("filter", filterValue);
+        } else {
+            sqlQuery = String.format("SELECT %s FROM %s", columnName, metadataTable);
+        }
+
+        getJdbcTemplate().query(sqlQuery, param, rs -> {
```

Comment:
Can please say how is DatabaseMetadataRetriever related to ArcSDEJdbcConnection ?

Reply:
+1 for deprecating ArcSDE

Reply:
Sure, but this will not be part of this pull request. That's for another pull request.